packages/smooth_app/lib/cards/category_cards/svg_cache.dart

```diff
@@ -80,14 +81,56 @@ class SvgCache extends AbstractCache {
       'nova-group-3.svg' => localizations.nova_group_3,
       'nova-group-4.svg' => localizations.nova_group_4,
       'nova-group-unknown.svg' => localizations.nova_group_unknown,
-      'nutriscore-a.svg' => localizations.nutriscore_a,
-      'nutriscore-b.svg' => localizations.nutriscore_b,
-      'nutriscore-c.svg' => localizations.nutriscore_c,
-      'nutriscore-d.svg' => localizations.nutriscore_d,
-      'nutriscore-e.svg' => localizations.nutriscore_e,
-      'nutriscore-unknown.svg' => localizations.nutriscore_unknown,
-      'nutriscore-not-applicable.svg' => localizations.ecoscore_not_applicable,
+      String _ when fileName.startsWith('nutriscore-') =>
+        _extractNutriScore(localizations, fileName),
       _ => null,
     };
   }
+
+  static String _extractNutriScore(
+    AppLocalizations localizations,
+    String fileName,
+  ) {
+    // Old NutriScore
+    if (fileName == 'nutriscore-a.svg') {
+      return localizations.nutriscore_a;
+    } else if (fileName == 'nutriscore-b.svg') {
+      return localizations.nutriscore_b;
+    } else if (fileName == 'nutriscore-c.svg') {
+      return localizations.nutriscore_c;
+    } else if (fileName == 'nutriscore-d.svg') {
+      return localizations.nutriscore_d;
+    } else if (fileName == 'nutriscore-e.svg') {
+      return localizations.nutriscore_e;
+    } else if (fileName == 'nutriscore-unknown.svg') {
+      return localizations.nutriscore_unknown;
+    } else if (fileName == 'nutriscore-not-applicable.svg') {
+      return localizations.nutriscore_not_applicable;
+    }
+
+    // NutriScore V2
+    if (fileName == 'nutriscore-unknown-') {
+      return localizations.nutriscore_unknown_new_formula;
+    } else if (fileName == 'nutriscore-not-applicable-') {
+      return localizations.nutriscore_not_applicable_new_formula;
+    } else {
```

Comment:
Just a matter of personal taste: as you `return` you don't need an `else`. Easier to read when there are no useless blocks.